security/pfSense-pkg-snort/files/usr/local/www/snort/snort_interfaces_global.php

```diff
@@ -88,68 +78,69 @@
+		config_set_path('installedpackages/snortglobal/openappid_detectors', $_POST['openappid_detectors'] ? 'on' : 'off');
+		config_set_path('installedpackages/snortglobal/openappid_rules_detectors',  $_POST['openappid_rules_detectors'] ? 'on' : 'off');
+		config_set_path('installedpackages/snortglobal/hide_deprecated_rules', $_POST['hide_deprecated_rules'] ? 'on' : 'off');
+		config_set_path('installedpackages/snortglobal/curl_no_verify_ssl_peer', $_POST['curl_no_verify_ssl_peer'] ? 'on' : 'off');
 
 		// If any rule sets are being turned off, then remove them
 		// from the active rules section of each interface.  Start
 		// by building an arry of prefixes for the disabled rules.
 		$disabled_rules = array();
 		$disable_ips_policy = false;
-		if ($config['installedpackages']['snortglobal']['snortdownload'] == 'off') {
+		if (config_get_path('installedpackages/snortglobal/snortdownload') == 'off') {
 			$disabled_rules[] = VRT_FILE_PREFIX;
 			$disable_ips_policy = true;
 		}
-		if ($config['installedpackages']['snortglobal']['snortcommunityrules'] == 'off')
+		if (config_get_path('installedpackages/snortglobal/snortcommunityrules') == 'off')
 			$disabled_rules[] = GPL_FILE_PREFIX;
-		if ($config['installedpackages']['snortglobal']['emergingthreats'] == 'off')
+		if (config_get_path('installedpackages/snortglobal/emergingthreats') == 'off')
 			$disabled_rules[] = ET_OPEN_FILE_PREFIX;
-		if ($config['installedpackages']['snortglobal']['emergingthreats_pro'] == 'off')
+		if (config_get_path('installedpackages/snortglobal/emergingthreats_pro') == 'off')
 			$disabled_rules[] = ET_PRO_FILE_PREFIX;
-		if ($config['installedpackages']['snortglobal']['openappid_rules_detectors'] == 'off')
+		if (config_get_path('installedpackages/snortglobal/openappid_rules_detectors') == 'off')
 			$disabled_rules[] = OPENAPPID_FILE_PREFIX;
-		if ($config['installedpackages']['snortglobal']['enable_feodo_botnet_c2_rules'] == 'off')
+		if (config_get_path('installedpackages/snortglobal/enable_feodo_botnet_c2_rules') == 'off')
 			$disabled_rules[] = "feodotracker";
 
 		// Now walk all the configured interface rulesets and remove
 		// any matching the disabled ruleset prefixes.
-		if (is_array($config['installedpackages']['snortglobal']['rule'])) {
-			foreach ($config['installedpackages']['snortglobal']['rule'] as &$iface) {
-				// Disable Snort IPS policy if Snort Subscriber rules are disabled
-				if ($disable_ips_policy) {
-					$iface['ips_policy_enable'] = 'off';
-					unset($iface['ips_policy']);
-				}
-				$enabled_rules = explode("||", $iface['rulesets']);
-				foreach ($enabled_rules as $k => $v) {
-					foreach ($disabled_rules as $d)
-						if (strpos(trim($v), $d) !== false)
-							unset($enabled_rules[$k]);
-				}
-				$iface['rulesets'] = implode("||", $enabled_rules);
+		$a_rules = config_get_path('installedpackages/snortglobal/rule', []);
+		foreach ($a_rules as $iface) {
```

Comment:
```suggestion
		foreach ($a_rules as &$iface) {
```
You still want the reference so you're modifying $a_rules

Reply:
Good catch! Missed that in my previous review. Did indeed need to be using the reference pointer.